### PR TITLE
PTX-15283 Improve validation for PX pods and add automation test case to cover issue that was found in CEE-744

### PR DESCRIPTION
**Pull request by nikolaypopov**
Improve validation for PX pods and add automation test case to cover issue that was found in CEE-744

Test case is to restrict from PX pods to be scheduled on a specific node due to resource limit (CPU) and then remove load to unblock CPU limit and validate PX is getting provisioned on the node as expected
Also some validation improvements

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: test/integration_test/regression_test.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+		// Validate StorageCluster is failing
+		if err := testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true); err == nil {
+			require.NoError(t, fmt.Errorf("Was expecting to fail validation deployment, but it was successful"))
+		}
+		logrus.Infof("Validation for StorageCluster [%s] failed as expected", cluster.Name)
+
+		// Get more info PX pod on a node we deploying apps on, PX pod shouldn't be provisioning due to 'Insufficient cpu'
+		err = validatePxPodOnNodeIsOutOfCpu(nodeNameWithLabel, cluster.Namespace)
+		require.NoError(t, err)
```
Comment (pure-jliao) [inline test/integration_test/regression_test.go:93]: should we validate rest of the storage pods are all healthy using `validateStorageClusterPods` function?

File: test/integration_test/regression_test.go [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+			},
+		}
+		cluster.Spec.Resources = resources
+
+		// Deploy StorageCluster
+		logrus.Infof("Deploying StorageCluster once [%s]", cluster.Name)
+		cluster = ci_utils.DeployStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate StorageCluster is failing
+		if err := testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true); err == nil {
```
Comment (pure-jliao) [inline test/integration_test/regression_test.go:86]: do you think the timeout is too long here? if we are going to validate the pod out of cpu, probably we can skip this? and validate other pods are healthy underneath except one.

File: test/integration_test/regression_test.go (hunk 1 above)
Comment (nikolaypopov) [reply test/integration_test/regression_test.go:93]: Done

File: test/integration_test/regression_test.go (hunk 2 above)
Comment (nikolaypopov) [reply test/integration_test/regression_test.go]: Done

File: test/integration_test/regression_test.go [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+		TestrailCaseIDs: []string{""},
+		TestSpec: func(t *testing.T) interface{} {
+			// Construct StorageCluster
+			cluster := &corev1.StorageCluster{}
+			cluster.Name = clusterName
+			err := ci_utils.ConstructStorageCluster(cluster, ci_utils.PxSpecGenURL, ci_utils.PxSpecImages)
+			require.NoError(t, err)
+			return cluster
+		},
+		TestFunc: NodeOutOfCpu,
```
[the remark this answers is not in the document]
Comment (nikolaypopov) [reply test/integration_test/regression_test.go:39]: I don't think we need to skip for 1.10.2 and below, this should fail the test and get us the expected result

File: test/integration_test/regression_test.go [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+	},
+}
+
+func shouldSkipRegressionTests(tc *types.TestCase) bool {
```
[the remark this answers is not in the document]
Comment (nikolaypopov) [reply test/integration_test/regression_test.go:43]: Removed

File: test/integration_test/utils/storagecluster.go [hunk 5]
```diff
@@ -161,6 +161,32 @@ func CreateStorageCluster(cluster *corev1.StorageCluster) (*corev1.StorageCluste
...
+	// Get the latest version of StorageCluster
+	liveCluster, err := operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	require.NoError(t, err)
+
+	return liveCluster
+}
+
 // DeployAndValidateStorageCluster creates and validates the storage cluster
 func DeployAndValidateStorageCluster(cluster *corev1.StorageCluster, pxSpecImages map[string]string, t *testing.T) *corev1.StorageCluster {
 	// Populate default values to empty fields first
```
[the remark this answers is not in the document]
Comment (nikolaypopov) [reply test/integration_test/utils/storagecluster.go:192]: Done

File: test/integration_test/utils/storagecluster.go [hunk 6]
```diff
@@ -161,6 +161,32 @@ func CreateStorageCluster(cluster *corev1.StorageCluster) (*corev1.StorageCluste
...
+		cluster = existingCluster
+		// Existing cluster from previous install
+		return cluster
+	}
+
+	// Get the latest version of StorageCluster
+	liveCluster, err := operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	require.NoError(t, err)
+
+	return liveCluster
```
[the remark this answers is not in the document]
Comment (nikolaypopov) [reply test/integration_test/utils/storagecluster.go:187]: Removed

**Review by pure-jliao — APPROVED** [review]
LGTM!

**Pull request merged**